src/errors.rs

```diff
@@ -444,6 +444,10 @@ fn component_unavailable_msg(cs: &[Component], manifest: &Manifest, toolchain: &
 }
 
 fn install_msg(bin: &str, toolchain: &str, is_default: bool) -> String {
+    if Toolchain::is_custom_name(toolchain) {
+        return format!("\nnote: this is a custom toolchain, which cannot use `rustup component add`\n\
+        help: if you installed this toolchain from source with `rustup component link`, you can build the component with `x.py`");
```

Comment:
```suggestion
        help: if you built this toolchain from source, and used `rustup component link`, then you may be able to build the component with `x.py`");
```

Basically make it less "yes you can do this"